### PR TITLE
Add detailed device name for user agent

### PR DESCRIPTION
**Pull request by h2kuo**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @h2kuo has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/Categories/UIDevice+BOXAdditions.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+//  Created on 3/21/16.
+//  Copyright © 2016 Box. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+@interface UIDevice (BOXAdditions)
+
+- (NSString *)detailedModelName;
+- (NSString *)modelID;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Categories/UIDevice+BOXAdditions.h]: This doesn't need to be public, right?

File: BoxContentSDK/BoxContentSDK/Requests/BOXRequest.m [hunk 2]
```diff
@@ -13,8 +13,8 @@
 #import "BOXISO8601DateFormatter.h"
 #import "BOXAppUserSession.h"
 #import "NSString+BOXAdditions.h"
+#import "UIDevice+BOXAdditions.h"
 #include <sys/types.h>
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXRequest.m]: Still need this?

File: BoxContentSDK/BoxContentSDK/Requests/BOXRequest.m [hunk 3]
```diff
@@ -404,21 +404,9 @@ - (NSString *)fullCollaborationFieldsParameterString
     return [array componentsJoinedByString:@","];
 }
 
-- (NSString *)modelID
+- (NSString *)deviceName
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXRequest.m]: Maybe deviceModelName?

deviceName is more when you set it to say "Helen's iPhone"

**Comment by boxtcarpel** [comment]
+2 LGTM

Some comments. Why did the CI builder fail?

:shell:

**Comment by h2kuo** [comment]
FYI @crousselle

**Comment by crousselle** [comment]
+2

**Pull request merged**